cachi2/core/package_managers/bundler/main.py

```diff
@@ -24,5 +31,49 @@ def fetch_bundler_source(request: Request) -> RequestOutput:
 
 def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
     """Process a request for a single bundler package."""
-    parse_lockfile(package_dir)
+    dependencies = parse_lockfile(package_dir)
+    _get_package_metadata(package_dir, dependencies)
     return []
+
+
+def _get_package_metadata(
+    package_dir: RootedPath,
+    dependencies: ParseResult,
+) -> tuple[str, Optional[str]]:
+    """Get package metadata - name and version."""
+    path_dependencies = [dep for dep in dependencies if isinstance(dep, PathDependency)]
+    metadata = _extract_metadata_from_gemspec_file(path_dependencies)
+    if metadata is not None:
+        return metadata
+
+    # fallback to origin remote
+    try:
+        name = get_repo_name_from_origin_remote(package_dir)
+    except UnsupportedFeature:
+        raise PackageRejected(
+            reason="Failed to extract package name from the git repository",
+            solution=(
+                "Please specify package metadata in a way that Cachi2 understands,\n"
+                "or make sure that the directory Cachi2 is processing is a git repository with\n"
+                "an 'origin' remote, in which case Cachi2 will infer the package name from the remote URL."
+            ),
+        )
+
+    return name, None
+
+
+def _extract_metadata_from_gemspec_file(
+    path_dependencies: list[PathDependency],
+) -> Optional[tuple[str, str]]:
+    """
+    Extract metadata from the gemspec file.
+
+    If the gemspec file exists in the package directory and is explicitly defined in the Gemfile,
+    then the Gemfile.lock will contain a path dependency representing the gem in the package directory.
+    """
+    log.debug("Extracting metadata from gemspec file")
```

Comment:
Nitpick: I think it is more useful to get a log stating that the info was not found in a gemspec file than just a generic log that traces that this function was called.